torch_geometric/nn/conv/ker_gnn_conv.py

```diff
@@ -0,0 +1,138 @@
+        self.kernel = kernel
+        self.power = power
+        self.size_graph_filter = size_graph_filter
+        self.size_subgraph = size_subgraph
+
+        if hidden_channels:
+            self.linear = Linear(in_channels, hidden_channels)
+            self.filter_x = torch.empty(size_graph_filter, hidden_channels,
+                                        out_channels)
+        else:
+            self.filter_x = torch.empty(size_graph_filter, in_channels,
+                                        out_channels)
+
+        self.filter_edge_index = torch.triu_indices(size_graph_filter,
+                                                    size_graph_filter, 1)
+        self.filter_edge_attr = torch.empty(
+            size_graph_filter * (size_graph_filter - 1) // 2, out_channels)
+        self.filter_g = None
+
+        self.simple_conv = SimpleConv()
+        self.dropout = nn.Dropout(p=dropout)
+        self.relu = nn.ReLU()
+        self.ker_linear = Linear(
+            out_channels,
+            out_channels) if kernel == 'drw' else torch.nn.Identity()
+
+        self.reset_parameters()
+        assert self.filter_g is not None and self.filter_g.validate()
+
+    def reset_parameters(self):
+        super().reset_parameters()
+        self.filter_x.data.uniform_(0, 1)
+        self.filter_edge_attr.data.uniform_(-1, 1)
+
+        self.filter_edge_index, self.filter_edge_attr = to_undirected(
+            self.filter_edge_index, self.filter_edge_attr)
+
+        self.filter_g = Data(x=self.filter_x,
+                             edge_index=self.filter_edge_index,
+                             edge_attr=self.filter_edge_attr)
+
+    def forward(self, x: Union[Tensor, PairTensor], edge_index: Adj,
+                edge_weight: OptTensor = None) -> Tensor:
+        self.filter_g.to(x.device)
+        if self.hidden_channels:
+            x = self.relu(self.linear(x))
+
+        outs = []
+        z = self.filter_g.x
+        xz = x @ z
+        for i in range(self.power):
+            if i == 0:
+                o = self.propagate(edge_index, x=x, edge_weight=edge_weight,
+                                   z=z)
+            else:
+                x = self.simple_conv(x, edge_index=edge_index)
+                # propagate_type: (x: Tensor, edge_weight: OptTensor, z: Tensor, is_kernel: bool)
+                z = self.propagate(self.filter_g.edge_index,
+                                   x=z.reshape(self.size_graph_filter,
+                                               -1), edge_weight=edge_weight,
+                                   z=self.filter_g.edge_attr, is_kernel=True)
+                z = z.reshape(self.size_graph_filter, -1, self.out_channels)
+                o = self.propagate(edge_index, x=x, edge_weight=edge_weight,
+                                   z=z)
+            t = o * xz
+            outs.append(self.ker_linear(t))
+        return torch.mean(sum(outs) / len(outs), dim=[0])
```

Comment:
Since we are directly summing `outs`, can we use `outs += ...` instead of `outs.append(...)`?